src/FSharpVSPowerTools.Logic/Linting/OptionsViewModel.fs

```diff
@@ -0,0 +1,218 @@
+﻿namespace FSharpVSPowerTools.Linting
+
+open System
+open System.ComponentModel
+open System.Collections.ObjectModel
+open FSharpLint.Framework.Configuration
+open FSharp.ViewModule
+
+type BoolViewModel(name, isChecked) as this =
+    inherit ViewModelBase()
+
+    let isChecked = this.Factory.Backing(<@ this.IsChecked @>, isChecked)
+
+    member this.Name
+        with get() = name
+
+    member this.IsChecked
+        with get() = isChecked.Value
+        and set(v) = isChecked.Value <- v
+
+type IntViewModel(name, initialValue) as this =
+    inherit ViewModelBase()
+
+    let value = this.Factory.Backing(<@ this.Value @>, initialValue)
+
+    member this.Name
+        with get() = name
+
+    member this.Value
+        with get() = value.Value
+        and set(v) = value.Value <- v
+
+type HintsViewModel(config:Configuration) as this =
+    inherit ViewModelBase()
+
+    let validateHint hint =
+        [
+            if hint <> "" then
+                match FParsec.CharParsers.run FSharpLint.Framework.HintParser.phint hint with
+                | FParsec.CharParsers.Success(_) -> ()
+                | FParsec.CharParsers.Failure(message, _, _) -> yield message
+        ]
+
+    let newHint = this.Factory.Backing(<@ this.NewHint @>, "", validateHint)
+    let selectedHintIndex = this.Factory.Backing(<@ this.SelectedHintIndex @>, 0)
+
+    let hintSettings =
+        config.Analysers 
+            |> Seq.tryFind (fun x -> x.Key = "Hints")
+            |> (function 
+                | Some(x) -> x.Value.Settings |> Seq.map (fun x -> Some(x.Value)) 
+                | None -> Seq.empty)
+
+    let hints =
+        hintSettings 
+            |> Seq.tryPick (function | Some(Hints(hints)) -> Some(hints) | _ -> None)
+            |> (function | Some(x) -> ObservableCollection<_>(x) | None -> ObservableCollection<_>())
+
+    let isEnabled =
+        hintSettings 
+            |> Seq.tryPick (function | Some(Enabled(enabled)) -> Some(enabled) | _ -> None)
+            |> (function | Some(enabled) -> enabled | None -> false)
+
+    let isEnabled = this.Factory.Backing(<@ this.IsEnabled @>, isEnabled)
+
+    member this.Hints with get() = hints
+
+    member this.IsEnabled
+        with get() = isEnabled.Value
+        and set(v) = isEnabled.Value <- v
+
+    member this.AddHintCommand = 
+        this.Factory.CommandSync(fun _ -> 
+            if String.IsNullOrEmpty this.NewHint |> not then
+                match FParsec.CharParsers.run FSharpLint.Framework.HintParser.phint this.NewHint with
+                | FParsec.CharParsers.Success(_) -> 
+                    hints.Add(this.NewHint)
+                    this.NewHint <- ""
+                    this.SelectedHintIndex <- hints.Count - 1
+                | FParsec.CharParsers.Failure(_) -> ())
+
+    member this.RemoveHintCommand = 
+        this.Factory.CommandSyncParam(fun (selectedItem:obj) ->
+            if selectedItem :? string then
```

Comment:
If we use `match selectItem with :? string as s` then `s` is string and we don't need downcasting (`:?>`) anymore?